src/components/jellyfinApi.ts

```diff
@@ -11,28 +11,55 @@ export abstract class JellyfinApi {
     // Address of server
     public static serverAddress: string | null = null;
 
+    // device name
+    public static deviceName = 'Google Cast';
+
+    // unique id
+    public static deviceId = '';
+
+    // version
+    public static versionNumber = RECEIVERVERSION;
+
     public static setServerInfo(
         userId: string,
         accessToken: string,
-        serverAddress: string
+        serverAddress: string,
+        receiverName: string
     ): void {
         console.debug(
-            `JellyfinApi.setServerInfo: user:${userId}, token:${accessToken}, server:${serverAddress}`
+            `JellyfinApi.setServerInfo: user:${userId}, token:${accessToken}, ` +
+                `server:${serverAddress}, name:${receiverName}`
         );
         this.userId = userId;
         this.accessToken = accessToken;
         this.serverAddress = serverAddress;
+
+        // remove special characters from the receiver name
+        receiverName = receiverName.replace(/[^\w\s]/gi, '');
+
+        if (receiverName) {
+            this.deviceName = receiverName;
+            // deviceId just needs to be unique-ish
+            this.deviceId = btoa(receiverName);
+        } else {
+            const senders =
+                cast.framework.CastReceiverContext.getInstance().getSenders();
+
+            this.deviceName = 'Google Cast';
```

Comment:
Non blocking, but there's any chance this can be the real device name? i.e Chromecast Ultra, JBL, Chromecast Audio...

Reply:
I doubt it especially since groups are a thing, can be generated on the fly, and the device can be switched from the Google Home platform.

Reply:
Yeah but that shouldn't be a problem when the API exposes that?

The question is if the Cast API does.

Other wrappers like pyChromecast from Home Assistant does for example, but all of them are "clients", not "servers" like in this case.

Reply:
components/castDevices.ts uses codecs to figure out what type of chromecast we are dealing with, but not to the most precise degree. However, this is mostly a placeholder.

You cannot really figure out the identity of the chromecast from within this javascript, so what is done is that the cast client passes on the name that the device should take on `if (receiverName)`.

I think it's intentional that google does not expose unique identifiers to let developers recognize the same cast device.. https://stackoverflow.com/questions/23659927/is-there-a-way-to-get-a-unique-id-to-identify-a-chromecast-device


See also this that more directly answers the question: https://stackoverflow.com/questions/41706028/how-to-detect-chromecast-model-in-receiver

It's provided on the initiator side (your phone). So we have to get the ID sent to us.

Reply:
@hawken93 I see, thank you for the detailed explanation 👍